### ISSUE TITLE
Use expectations outside of a spec (or how to reuse expectation code)

### ISSUE THREAD

**Issue by jgongo** [kiwi-bdd/Kiwi#138]
Is there any way to use expectations outside of a spec? I had a lengthy test with several expectations that I wanted to factor out to use it in several examples. I have tried the following with varied success:
- Extract the expectations to a class method. I got the following error: No known class method for selector 'addMatchVerifierWithExpectationType:callSite:'
- Extract the expectations to an instance method. I got the following error: No visible @interface for 'NameOfClassWithCodeToReuse' declares the selector 'addMatchVerifierWithExpectationType:callSite:'
- Extract the expectations to a static inline function. I got the following error: Use of undeclared identifier 'self'
- Extract the expectations to a parameterized macro. This way works but XCode shows the errors in the line where the macro is used, so I don't know which expectation is failing.
- Extract the expectations to a block. This way seems to work, getting in XCode the name of the example that failed and the expectation that failed.

So summing up, is there any recommended way to reuse expectation code in several examples / specs?

Sorry if this is not the right place to ask this... any way, if I get an answer I'll be glad to update the wiki with this information. 

Thanks a lot
